src/compiler/checker.ts

```diff
@@ -393,9 +398,28 @@ namespace ts {
             }
 
             let sourceFiles = host.getSourceFiles();
-            return sourceFiles.indexOf(file1) <= sourceFiles.indexOf(file2);
-        }
+            return sourceFiles.indexOf(referenceDeclarationFile) <= sourceFiles.indexOf(locationNodeFile) || !isInSameLexicalScope(); 
+
+            function isInSameLexicalScope() {
+                let referenceDeclarationScope = getEnclosingBlockScopeContainer(referenceDeclaration);

```

Comment:
why are we using the block scopes? this is already handled in the look up. so what we care about is just function boundary.. so just getContainingFunction should do.
so should not it just be `getContianignFunction(referenceDeclaration) === getContainignFunction(locationNode)`?